server/src/domain/user/user.service.ts

```diff
@@ -93,10 +93,25 @@ export class UserService {
     authUser: AuthUserDto,
     fileInfo: Express.Multer.File,
   ): Promise<CreateProfileImageResponseDto> {
+    const user = await this.findOrFail(authUser.id, { withDeleted: true });
     const updatedUser = await this.userRepository.update(authUser.id, { profileImagePath: fileInfo.path });
+    if (user.profileImagePath !== '') {
```

Comment:
How or why would this ever be an empty string when using "createProfileImage"?

Reply:
If a user has never set is profile image, the default value is an empty string

Reply:
Oh, I see. What do you think about having it be a static path instead of a random uuid and then having to delete the old one? So overwrite instead of new + delete.

Reply:
I don't follow. How do you determine if a user has set its profile picture ?

Reply:
Yeah it should work. Should we use the user id for the static path ?

Reply:
Maybe we should change the profileImagePath to be a boolean

Reply:
Ah, hold on. I guess the extension may not always be the same, so you will have to potentially delete it afterwards, so maybe it is fine like you have it.

Reply:
I did get confused with user vs updated user and the user.profilePath is the old path, but referenced after you updated to the new one. Maybe you could do something like `const { profilePath: oldPath } = await this.findOrFail(...)`; to make it a little more clear.

Reply:
Done